microsoft/botframework-solutions
# [Virtual Assistant][TypeScript] Change i18n implementation to load different locales files

## Description
The previous implementation of internationalization needs to load the different locale files from a single directory. We change that library to `i18next` that allows us to load locale files from other directories.

## Testing Steps
> Prerequisites: `.bot` file and `LocaleConfigurations/*.bot` files with the services needed for the bot.
1. `npm install -g @microsoft/rush` to install the build tool.
1. Add `.bot` file path and secret to `./solutions/Virtual-Assistant/src/typescript/assistant/.env.development`.
1. Add `LocaleConfigurations/*.bot` paths to `./solutions/Virtual-Assistant/src/typescript/assistant/src/languageModels.json`.
1. `rush update` to install dependencies and symlink packages.
1. `rush rebuild` to build the solution.
1. `rush start` to run the bot.
1. Open the Bot Framework Emulator and set the locale to `en`.
1. Send `help` -> the bot should responds with `Help for Virtual Assistant`.
1. Change the locale to `es`.
1. Send `ayuda` -> the bot should responds with `Ayuda para el asistente virtual`.


## Checklist
